edb/edgeql/compiler/stmt.py

```diff
@@ -235,6 +235,26 @@ def compile_ForQuery(
     return result
 
 
+@dispatch.compile.register(qlast.InternalGroupQuery)
+def compile_InternalGroupQuery(
+    expr: qlast.InternalGroupQuery, *, ctx: context.ContextLevel
+) -> irast.Set:
+    # We want to disallow use of FOR GROUP except for when running
+    # a dev build. Check for the presence of a test schema element
+    # to detect whether we're on a dev build.
+    if not expr.from_desugaring and not ctx.env.schema.get(
+        'cfg::TestSessionConfig', None
```

Comment:
I'd be more rigorous and pass the `__internal_testmode` config value in `CompilerOptions` from `_compile_ql_query` instead.

Reply:
That's a pain for local development, though, because `__internal_testmode` can only be set per-session

Reply:
Fine, move the test schema hack up there then.  I'd still like `testmode` to show up as a `CompilerOptions` flag.

Reply:
Yeah good point